demos/common/cpp/models/src/classification_model.cpp

```diff
@@ -158,19 +158,16 @@ void ClassificationModel::prepareInputsOutputs(std::shared_ptr<ov::Model>& model
     model = ppp.build();
 
     // --------------------------- Adding softmax and topK output  ---------------------------
-    auto nodes = model->get_ops();
-    auto softmaxNodeIt = std::find_if(std::begin(nodes), std::end(nodes), [](const std::shared_ptr<ov::Node>& op) {
-        return std::string(op->get_type_name()) == "Softmax";
-    });
-
-    std::shared_ptr<ov::Node> softmaxNode;
-    if (softmaxNodeIt == nodes.end()) {
-        auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
-        softmaxNode = std::make_shared<ov::op::v1::Softmax>(logitsNode->output(0), 1);
-    } else {
-        softmaxNode = *softmaxNodeIt;
+
+    if (model->get_output_size() != 1) {
+        throw std::logic_error("Expected model's number of output equal to 1 (but found: " + std::to_string(model->get_output_size()) + ")");
     }
+
+    auto logitsNode = model->get_output_op(0)->input(0).get_source_output().get_node();
+
     const auto k = std::make_shared<ov::op::v0::Constant>(ov::element::i32, ov::Shape{}, std::vector<size_t>{nTop});
+    std::shared_ptr<ov::Node> softmaxNode = std::make_shared<ov::op::v1::Softmax>(logitsNode->output(0), 1);
```

Comment:
Why would you move SoftMax construction to the middle of area that constructs topk. Move it above.

Reply:
Guess it's about the order of lines